src/Game/Game.elm

```diff
@@ -115,7 +115,10 @@ update msg model =
             ( { model | currentScreen = InventoryScreen }, Cmd.none )
 
         InvMsg msg ->
-            ( Inventory.update msg model, Cmd.none )
+            if msg == ExitMsg then
+                ( { model | currentScreen = MapScreen }, Cmd.none )
```

Comment:
Okay... so the way Elm composes components is the outer component (Game) will include the inner component (Inventory). To do this, the Inventory messages get wrapped in Game messages and unwrapped when the Inventory wants to work with them. That way every message in the app is actually a Game message, even though they contain inner Inventory messages. So, what we want to do is pass an action value into Inventory and have it be able to accept it so it can embed it in its view, but without knowing what it is beforehand.

So, the view function in Inventory needs to take a first argument. That will be the "message result" (ie action) value that we want to embed. When we embed our Inventory.view in our Game.view, it'll take this Game exit message action as its first argument. The Inventory.view won't know anything about it other than it'll be some HTML attribute producing action value. In the Game.view we pass the action we want to be attached to the button in the Inventory.view.

Hopefully this makes sense.

This is one way to do it. It's useful to experiment while trying to keep the idea of not letting either of the components know about the other one's internals, just use the API that they each expose. This will be your teacher.